services/secondary-storage/controller/src/org/apache/cloudstack/secondarystorage/SecondaryStorageManagerImpl.java

```diff
@@ -1118,7 +1119,7 @@ public boolean finalizeVirtualMachineProfile(VirtualMachineProfile profile, Depl
 
         StringBuilder buf = profile.getBootArgsBuilder();
         buf.append(" template=domP type=secstorage");
-        buf.append(" host=").append(ApiServiceConfiguration.ManagementHostIPAdr.value());
+        buf.append(" host=").append(StringUtils.shuffleCSVList(ApiServiceConfiguration.ManagementHostIPAdr.value()));
```

Comment:
Yes, we want to have some randomness in case of multiple-mgmt server ips (if configured) that's still better than a static/configured value.